### PR TITLE
adding gatekeeper deployment

### PR DESCRIPTION
**Pull request by marzianor9**

### CODE REVIEW

File: feature-configs/deploy/gatekeeper/mutations.yaml [hunk 1]
```diff
@@ -0,0 +1,1446 @@
+---
```
Comment (SchSeba) [inline feature-configs/deploy/gatekeeper/mutations.yaml:1]: please split this file into
1. CRD
2. Namespace
3. RBAC (roles and service accounts)
4. deployment + service
5. webhook

**Review by SchSeba — COMMENTED** [review]
thanks for the PR!

Please add a `is_ready.sh` so we can be sure everything is ready when we deploy the feature.
also this feature can be tested in the CI so please add `gatekeeper` in the Makefile line 2 export.

Last request can you add ontop of this commit the PR with the functional tests I will like to see them running before we merge this one (we can remove the commit after that)

File: feature-configs/deploy/gatekeeper/services.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
```
Comment (SchSeba) [inline feature-configs/deploy/gatekeeper/services.yaml:1]: move this one to the webhook please.

**Review by SchSeba — COMMENTED** [review]
This looks better but you didn't create the is_ready and change the Makefile as requested

File: feature-configs/deploy/gatekeeper/is_ready.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+oc -n gatekeeper-system wait deployment/gatekeeper-controller-manager --for condition=available --timeout 5s
+oc -n gatekeeper-system wait deployment/gatekeeper-audit --for condition=available --timeout 5s
```
Comment (SchSeba) [inline feature-configs/deploy/gatekeeper/is_ready.sh:3]: nit: please add a new line here

File: feature-configs/deploy/gatekeeper/is_ready.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+oc -n gatekeeper-system wait deployment/gatekeeper-controller-manager --for condition=available --timeout 5s
```
Comment (SchSeba) [inline feature-configs/deploy/gatekeeper/is_ready.sh:2]: please wait for only 1s here the deploy loop will retry until it's ready so no need to wait 5s here

File: functests/test.go [hunk 5]
```diff
@@ -1 +0,0 @@
-package test
```
Comment (SchSeba) [inline functests/test.go:1]: why you remove this?

File: feature-configs/deploy/gatekeeper/kustomization.yaml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
```
Comment (SchSeba) [inline feature-configs/deploy/gatekeeper/kustomization.yaml:3]: please make the file names singular not plural

```
namespaces -> namespace
```

File: feature-configs/deploy/gatekeeper/kustomization.yaml [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+  - namespaces.yaml
+  - crds.yaml
+  - webhooks.yaml
+  - rbac.yaml
+  - services.yaml
```
Comment (SchSeba) [inline feature-configs/deploy/gatekeeper/kustomization.yaml:8]: this one doesn't exist anymore

**Review by SchSeba — COMMENTED** [review]
few changes please

**Review by SchSeba — COMMENTED** [review]
/ok-to-test

**Comment by SchSeba** [comment]
/lgtm

/hold
I want to see the tests are running on the CI and then we are going to remove the test commit from here so we can review it on it's own PR.

**Comment by SchSeba** [comment]
The tests are running as expected

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-kni_cnf-features-deploy/404/pull-ci-openshift-kni-cnf-features-deploy-master-e2e-gcp-ovn/1346199948894408704/artifacts/e2e-gcp-ovn/e2e-gcp-ovn/cnftests-junit.xml

```
<testcase name="gatekeeper should be able to add metadata info(labels/annotations) test-pod-lab-added-mutation" classname="CNF Features e2e integration tests" time="1.3927550530000001"/>
<testcase name="gatekeeper should be able to add metadata info(labels/annotations) test-pod-ann-added-mutation" classname="CNF Features e2e integration tests" time="0.106004138"/>
<testcase name="gatekeeper should not mutate existing metadata info(labels/annotations) test-pod-lab-exist-mutation" classname="CNF Features e2e integration tests" time="0.083933671"/>
<testcase name="gatekeeper should not mutate existing metadata info(labels/annotations) test-pod-ann-exist-mutation" classname="CNF Features e2e integration tests" time="0.687675501"/>
<testcase name="gatekeeper should apply mutations by order" classname="CNF Features e2e integration tests" time="1.599889053"/>
<testcase name="gatekeeper should be able to update mutation policy" classname="CNF Features e2e integration tests" time="1.399841358"/>
<testcase name="gatekeeper should not apply mutations policies after deletion" classname="CNF Features e2e integration tests" time="1.198118965"/>
<testcase name="gatekeeper should be able to match by any match category" classname="CNF Features e2e integration tests" time="3.199322342"/>
```

**Comment by SchSeba** [comment]
@marzianor9 please remove the tests commit from this PR

**Comment by SchSeba** [comment]
/hold

@sabinaaledort @yuvalk can you please take a look I think this PR will require some change in the CI system?

**Comment by sabinaaledort** [comment]
> /hold
> 
> @sabinaaledort @yuvalk can you please take a look I think this PR will require some change in the CI system?

@SchSeba The only change i can think of is adding gatekeeper to the features env var in our deployment script.

**Comment by yuvalk** [comment]
> /hold
> 
> @sabinaaledort @yuvalk can you please take a look I think this PR will require some change in the CI system?

yeah, it was inevitable :-)

File: feature-configs/deploy/gatekeeper/deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+            - name: POD_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: metadata.namespace
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+          image: quay.io/mmirecki/gatekeeper_frankenstein
```
Comment (yuvalk) [inline feature-configs/deploy/gatekeeper/deployment.yaml:44]: is that the 'official' location for that?

**Comment by yuvalk** [comment]
this whole gatekeeper wont be an operator of it's own?
just a crd and some pods?

**Comment by SchSeba** [comment]
> this whole gatekeeper wont be an operator of it's own?
just a crd and some pods?

@yuvalk it will be in the future but for this release we want to test the gatekeeper without the need of operator that is still under development

File: feature-configs/deploy/gatekeeper/deployment.yaml (hunk 8 above)
Comment (SchSeba) [reply feature-configs/deploy/gatekeeper/deployment.yaml:44]: for now yes until the GK will have a new tag and image

File: feature-configs/deploy/gatekeeper/webhook.yaml [hunk 9]
```diff
@@ -0,0 +1,93 @@
+---
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: MutatingWebhookConfiguration
+metadata:
+  name: gatekeeper-mutating-webhook-configuration
+webhooks:
+  - clientConfig:
+      caBundle: Cg==
```
Comment (yuvalk) [inline feature-configs/deploy/gatekeeper/webhook.yaml:8]: that's a weird one
why do we want a newline there?

Comment (SchSeba) [reply feature-configs/deploy/gatekeeper/webhook.yaml:8]: good question maybe the deployment updates the CA after that?

@mmirecki can you help here?

Comment (mmirecki) [reply feature-configs/deploy/gatekeeper/webhook.yaml:8]: Gatekeeper will generate one.
Please check [here](https://github.com/open-policy-agent/cert-controller) for more details.

**Comment by fedepaol** [comment]
I wonder if now it makes sense to deploy using the operator
WDYT @mmirecki @marzianor9 @SchSeba  ?

**Comment by mmirecki** [comment]
> I wonder if now it makes sense to deploy using the operator
> WDYT @mmirecki @marzianor9 @SchSeba ?

Inded, the next PR will remove all the files in favor of the operator.
Initially we did not know when the operator will come alive, but now that it has I think it makes sense to skip this in favor of using the operator.

**Comment by SchSeba** [comment]
closing in favor of https://github.com/openshift-kni/cnf-features-deploy/pull/452

/close

**Pull request closed without merging**